libraries/AP_EFI/AP_EFI_Serial_Hirth.cpp

```diff
@@ -0,0 +1,328 @@
+                        }
+                    }
+
+                    res_data.checksum = port->read();
+                    computed_checksum = computed_checksum % 256;
+                    // computed_checksum = (256 - computed_checksum);
+                    // discard further bytes if checksum is not matching
+                    if (res_data.checksum != (CHECKSUM_MAX - computed_checksum)) {
+                        internal_state.crc_fail_cnt++;
+                        port->discard_input();
+                    }
+                    else {
+                        internal_state.uptime = now - last_uptime;
+                        last_uptime = now;
+                        internal_state.last_updated_ms = now;
+                        decode_data();
+                        copy_to_frontend();
+                    }
+                    waiting_response = false;
+                }
+            }
+        }
+        
+        //sending cmd
+        if(!waiting_response) {
+
+            // get new throttle value
+            new_throttle = (uint16_t)SRV_Channels::get_output_scaled(SRV_Channel::k_throttle);
+
+            // Limit throttle percent to 65% for testing purposes only
+            if (new_throttle > 65)
+            {
+                new_throttle = 65;
+            }
+
+            //check for change or timeout for throttle value
+            if ((new_throttle != old_throttle) || (now - last_req_send_throttle > 500)){
+                // if new throttle value, send throttle request
+                status = send_target_values(new_throttle);
+                old_throttle = new_throttle;
+                internal_state.k_throttle = new_throttle;
+                last_req_send_throttle = now;
+            }
+            else {
+                //request Status request, if no throttle commands
+                status = send_request_status();
+            }      
+
+            if (status == true) {
+                waiting_response = true;
+                last_response_ms = now;
+            }
+        }
+    }
+}
+
+
+/**
+ * @brief updates the current quantity that will be expected
+ * 
+ */
+void AP_EFI_Serial_Hirth::get_quantity() {
+    switch (req_data.code)
+    {
+    case CODE_REQUEST_STATUS_1:
+        expected_bytes = QUANTITY_RESPONSE_STATUS_1;
+        break;
+    case CODE_REQUEST_STATUS_2:
+        expected_bytes = QUANTITY_RESPONSE_STATUS_2;
+        break;
+    case CODE_REQUEST_STATUS_3:
+        expected_bytes = QUANTITY_RESPONSE_STATUS_3;
+        break;
+    case CODE_SET_VALUE:
+        expected_bytes = QUANTITY_ACK_SET_VALUES;
+        break;
+    }
+}
+
+
+/**
+ * @brief sends the new throttle command to Hirth ECU
+ * 
+ * @param thr - new throttle value given by SRV_Channel::k_throttle
+ * @return true - if success
+ * @return false - currently not implemented
+ */
+bool AP_EFI_Serial_Hirth::send_target_values(uint16_t thr) {
+    int idx = 0;
```

Comment:
`uint8_t` and move it closer to where it's being used

Reply:
modified code